src/scripts/main.js

```diff
@@ -1,5 +1,38 @@
 'use strict';
 
+const spider = document.querySelector('.spider');
+const wall = document.querySelector('.wall');
+// we create values from elements before point
+// when we set handler and use those as we need
+
 document.addEventListener('click', (e) => {
-  // write code here
+  const field = e.target.closest('.wall');
+
+  if (!field) {
+    return; // we get rid of options when "click" happens outside the wall
+  }
+
+  const wallRect = wall.getBoundingClientRect();
+  const spiderRect = spider.getBoundingClientRect();
+  const widthSpider = spider.offsetWidth; // the spider is square
+  const border = wall.clientTop;
+
+  let x = e.clientX - widthSpider / 2 - border - wallRect.left;
```

Comment:
The calculation for 'x' is a bit confusing. It might be helpful to add a comment explaining why you're subtracting half the spider's width and the border from the client's x position.